### PR TITLE
Add etherscan document loader

### PR DESCRIPTION
**Pull request by glazec**
@rlancemartin 
The modification includes:
* etherscanLoader
* test_etherscan
* document ipynb

I have run the test, lint, format, and spell check. I do encounter a linting error on ipynb, I am not sure how to address that.
```
docs/extras/modules/data_connection/document_loaders/integrations/Etherscan.ipynb:55: error: Name "null" is not defined  [name-defined]
docs/extras/modules/data_connection/document_loaders/integrations/Etherscan.ipynb:76: error: Name "null" is not defined  [name-defined]
Found 2 errors in 1 file (checked 1 source file)
```
- Description: The Etherscan loader uses etherscan api to load transaction histories under specific accounts on Ethereum Mainnet.
- No dependency is introduced by this PR.
- Twitter handle: glazecl

### CODE REVIEW

File: langchain/document_loaders/etherscan.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            "normal_transaction",
+            "internal_transaction",
+            "erc20_transaction",
+            "eth_balance",
+            "erc721_transaction",
+            "erc1155_transaction",
+        ]:
+            raise ValueError(f"Invalid filter {filter}")
+
+    def load(self) -> List[Document]:
```
Comment (rlancemartin) [inline langchain/document_loaders/etherscan.py:55]: Nice addition!

Nit: make this `lazy_load` and `yield` documents.

Then add as load method:
```
    def load(self) -> List[Document]:
        """Load Documents from table."""
        return list(self.lazy_load())
```

Useful if folks want to lazy load large volumes of data.

See example:
https://github.com/hwchase17/langchain/blob/master/langchain/document_loaders/airtable.py

**Comment by rlancemartin** [comment]
Also, see Lint:
```
poetry run ruff .
langchain/document_loaders/etherscan.py:72:89: E501 Line too long (240 > 88 characters)
langchain/document_loaders/etherscan.py:90:89: E501 Line too long (139 > 88 characters)
langchain/document_loaders/etherscan.py:99:89: E501 Line too long (248 > 88 characters)
langchain/document_loaders/etherscan.py:116:89: E501 Line too long (241 > 88 characters)
langchain/document_loaders/etherscan.py:133:89: E501 Line too long (244 > 88 characters)
langchain/document_loaders/etherscan.py:150:89: E501 Line too long (245 > 88 characters)
tests/integration_tests/document_loaders/test_etherscan.py:67:9: F841 [*] Local variable `loader` is assigned to but never used
```

Run `make format` to check prior to pushing.

**Comment by glazec** [comment]
@rlancemartin add the loader and fix the linter problem

**Comment by baskaryan** [comment]
awesome, thanks @glazec!

**Comment by baskaryan** [comment]
thanks @glazec!

**Pull request merged**